accepted/0000-salt-new-maintaining-workflow.md

```diff
@@ -0,0 +1,300 @@
+  * https://build.opensuse.org/package/show/home:PSuarezHernandez:tests:github:CI:meaksh:salt:PR-10/salt
+
+Feel free to open new PRs against `openSUSE/devel/master` to see this in action.
+
+### Salt Extensions
+
+#### Builtin extensions
+The sources for the builtin Salt Extensions will be located together with the main Salt codebase at the `openSUSE/salt` GitHub repository. No new packages or subpackages will be created for these extensions as they will be part of the main `python3*-salt` package.
+
+If a fix is needed for any of the builtin extensions, workflow would be the same as for a code fix in the main Salt package.
+
+#### Packaged Salt Extensions
+
+For the Salt Extensions that are packaged separately from the main Salt package, we will create a separated GitHub repository where we will maintain these extensions.
+
+This "openSUSE/salt-extensions" repository will contain:
+- a common salt-extension spec file that will generate all RPM packages
+- The sources for each Salt Extension we package
+- The changelog files for each maintained codestream
+- OBS workflow file
+
+When it comes to OBS, we will use the same SCM integration and OBS subprojects schema than the proposed for the main Salt codebase. Unique workflow for Salt and Salt Extensions.
+
+- PoC:
+  * https://github.com/meaksh/test-repo-1
+  * https://build.opensuse.org/package/show/home:PSuarezHernandez:tests/salt-extensions
+
+- Example PR:
+  * https://github.com/meaksh/test-repo-1/pull/6
+  * https://build.opensuse.org/project/show/home:PSuarezHernandez:tests:github:CI:meaksh:test-repo-1:PR-6
+
+NOTE: We are using a single GitHub repo and single OBS package which provides all different salt-extensions RPM packages. This is preferred against having a separated GitHub repositories and OBS package for each Salt Extension, as it will reduce the number of submissions, maintenance incidents and resources needed.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- A bit more complex OBS structure than the current one. Including `obs_scm` service.
+- Having to still rely on Jenkins to get the packages ready to be released.
+
+# Alternatives
+[alternatives]: #alternatives
+
+1. Stick to our current workflow based on "salt-packaging" -> The workflow doesn't currently fit with Salt Extensions and we don't want to have different workflows between Salt and Salt Extensions.
+2. One dedicated GitHub repository and OBS package per each Salt Extension -> It won't save resources and will cause more submissions.
+3. The usage of "git submodules" as an alternative to adding the Salt Extensions sources manually would make it tricky to generate patches manually and also to integrate with "obs_scm".
+4. Use OBS `scmsync` integration -> while this allows to integrate even the `_service` file into the GitHub repo, it does not work well with SCM/CI workflow, therefore we cannot test the build of the package for an open PR because the branched package is not reflecting the actual changes on the PR.
```

Comment:
The current approach uses a combination of Jenkins and the OBS<>Github integration, but we still rely on Jenkins as you point out in #Drawbacks. Do we win much by this setup? I wonder if a setup that does not use OBS' SCM/CI Github integration, one that just relies on Jenkins for triggering new builds, would be simpler (as in fewer moving parts). Long term, `scmsync` might be the way to go.